Lib/typing.py

```diff
@@ -3296,3 +3298,41 @@ def method(self) -> None:
         # read-only property, TypeError if it's a builtin class.
         pass
     return method
+
+
+def is_protocol(tp: type, /) -> bool:
+    """Return True if the given type is a Protocol.
+
+    Example::
+
+        >>> from typing import Protocol, is_protocol
+        >>> class P(Protocol):
+        ...     def a(self) -> str: ...
+        ...     b: int
+        >>> is_protocol(P)
+        True
+        >>> is_protocol(int)
+        False
+
+    """
+    return getattr(tp, '_is_protocol', False) and tp is not Protocol
```

Comment:
What about:

```suggestion
    return (
        isinstance(tp, _ProtocolMeta)
        and getattr(tp, '_is_protocol', False)
        and tp is not Protocol
    )
```

Reply:
We should ensure that this works well with `typing_extensions.Protocol` out of the box.

Reply:
I thought you might say that. Can we at least verify that it is a class we're dealing with?

```suggestion
    return (
        isinstance(tp, type)
        and getattr(tp, '_is_protocol', False)
        and tp is not Protocol
    )
```